cmd/wavelet/events.go

```diff
@@ -8,15 +8,15 @@ import (
 )
 
 // converts a normal (func to close, error) to only an error
-func addToCloser(toClose []func()) func(f func(), err error) error {
+func addToCloser(toClose *[]func()) func(f func(), err error) error {
 	return func(f func(), err error) error {
-		toClose = append(toClose, f)
+		*toClose = append(*toClose, f)
```

Comment:
I think there's potential bug that can cause panic.

In the closure, we should check argument `f` is not nil before appending.

This is because if the Poll failed to Dial, it'll return (nil, err).

Reply:
nice catch. fixed